Improve organizer/event-series calendar UI on mobile

wiffbi opened the pull request:
This PR improves accessibility on the mobile calendar-view as it turns the calendar-table into a role="grid" with keyboard-selectable option, info on number of events on that day and making `#selected-day` an aria-live-region.

It further improves the mobile UI by mimicking a single event box on the day-label with the most important event-availability taking precedence.

Before (just brand-primary, no hover, no selected state):
![Bildschirmfoto 2024-11-26 um 10 16 28](https://github.com/user-attachments/assets/669475c4-eef1-437a-b838-2514d51b7be9)

After:
![Bildschirmfoto 2024-11-26 um 10 26 18](https://github.com/user-attachments/assets/efb2da0d-5d99-498b-bd0d-9a8f503caf0d)

raphaelm commented:
Hmm. I'm not fully sold on the style with mimicking the event box. Your screenshot looks fine, but uses a very wide definition of "mobile". If I set my developer tools to "iphone SE" as the extreme case for a really small smartphone, I have a horizontal scrollbar in the calendar and sunday is cutoff. Also, the touch targets are really really small, I don't think I would hit them.

![image](https://github.com/user-attachments/assets/897c6736-aef5-41f9-a55c-502af30ee20a)

wiffbi commented on src/pretix/static/pretixpresale/js/ui/main.js:
```diff
@@ -459,10 +459,22 @@ $(function () {
+    }).each(function() {
+        // check all events classes and set the "winning" class for the availability of the day-label on mobile
+        var $dayLabel = $('.day-label', this);
+        var classes = ['available', 'waitinglist', 'soon', 'reserved', 'soldout', 'continued', 'over'];
```
I will look into this

wiffbi replied:
I made it so it works like the widget currently – only showing low if all available events are low. It would be a tiny change to make it so, that we add the low-indicator if any available is low.

raphaelm replied:
> The logic in the widget is the same as it shows the „best“ status available.

Okay, I was wondering if they have the same interpretation of what "best" is, but I'll take that as a yes ;)

> It adds the low-indicator only if all events are low. IMHO it feels more suitable to add that indicator if at least one event in the list is low – the „green“ informs that there are events available, so it is worth it to click. Adding the indicator if at least one is low also helps to increase the urgency to book now? Should I change that bevahiour (of course in the widget as well) or should I take the logic from the widget to only show low-availability if all are low?

No, I think it is correct the way it is in the widget. Let's say our day has 3 events and consider three scenarios:
- available + available + available → showing available, no argument there
- available + available + low → ?
- available + available + soldout → showing available, no argument there

If the middle scenario would show "low", it's a little weird that it will show "available" again in the third scenario. So I think showing "available" is the logical option.

wiffbi replied:
Okay, makes sense. I already made it that way in the code.

wiffbi commented on src/pretix/presale/templates/pretixpresale/fragment_calendar.html:
```diff
@@ -18,7 +18,25 @@
+                                {% else %}
+                                    <span aria-hidden="true" class="day-label">{{ day.day }}</span>
+                                {% endif %}
+                                    <span class="sr-only">
+                                        ({% blocktrans trimmed count count=day.events|length %}
+                                            {{ count }} event
+                                        {% plural %}
+                                            {{ count }} events
+                                        {% endblocktrans %})
+                                    </span>
```
Semantically speaking probably not, I will try to find another solution – when using the screenreader, it read the number of the day in table (hence the aria-hidden I added on there) and the date (from the date-attribute of the time-element), when tabbing through the days, which was kind of annoying. So I looked for a way to remove the number of the day, but keep VoiceOver reading the date. I will look into this, maybe instead add a „readable“ date as `.sr-only`.

wiffbi commented:
I changed the calendar to not use as many slightly different color-variations and instead use the alert-color-combinations (which now use the better shade/tint-color-mixings instead of lighten/darken) to make it look more consistent and ensure a good color-contrast.

For the low-availability IMHO the yellow bar to the right like in the widget does not work as well when using the alert-style-event-boxes (needed for a11y color contrasts), as that already has a thick bar to the left. So I added a yellow gradient into the green bar like so:

![Bildschirmfoto 2024-11-29 um 13 20 37](https://github.com/user-attachments/assets/6514b50c-15ec-4993-a7bf-5d7c07b660d9)
